### ISSUE TITLE
There is no implementation of the addition operation for compressed matrices

### ISSUE THREAD

**Issue by madc0ww** [viennacl/viennacl-dev#259]
Hi, I want to sum two **compressed matrices**, but it seems that there is no `operator+` or `operator+=` implemented for them:

```c++
viennacl::compressed_matrix<float> a(3, 3);
viennacl::compressed_matrix<float> b(3, 3);

// Considering it is not using GPU for this toy example, so I can allocate
// dynamic memory on the device
x(1,1) = 1.0f;
y(1,2) = 2.0f;

// The same occurs for x += y;
x = x + y;
```

I got the following error:

> error: no match for ‘operator+’ (operand types are ‘viennacl::compressed_matrix<float>’ and
> ‘viennacl::compressed_matrix<float>’)
>  91      x = x + y;
> 92              \~\~\^\~\~                                                                        

I'm using ViennaCL with **OpenCL**.

**Comment by karlrupp** [comment]
Yes, there is indeed no summation implemented yet. It will be added in the not too distant future.

**Comment by karlrupp** [comment]
Yes, there is indeed no summation implemented yet. It will be added in the not too distant future.

**Comment by madc0ww** [comment]
Thanks for the quick answer, @karlrupp

**Comment by madc0ww** [comment]
@karlrupp The methods `begin1()`, `begin2()`, `end1()` and `end2()` are not available for **compressed matrices**. Will it be added in the future too? Should I open an issue for that?
